### ISSUE TITLE
When a new custom field is added to Salesforce Platform Event, throwing uncaught exception that crash the Node.js process

### ISSUE THREAD

**Issue by bbuyukormeci** [pozil/pub-sub-api-node-client#16]
```
[2023-09-20T11:15:11.404Z] Received 1 events, latest replay ID: 4076479
[2023-09-20T11:15:11.418Z] [error] Worker 518d2ced-8ba2-42a2-ba77-2e98fb4c903d uncaught exception (learn more: https://go.microsoft.com/fwlink/?linkid=2097909 ): Error: trailing data
     at Type.fromBuffer (/Users/repo/pe-consumer/node_modules/avro-js/lib/schemas.js:252:11)
     at parseEvent (/Users/repo/pe-consumer/node_modules/salesforce-pubsub-api-client/dist/client.cjs:142:31)
     at /Users/repo/pe-consumer/node_modules/salesforce-pubsub-api-client/dist/client.cjs:559:33
     at Array.forEach (<anonymous>)
     at ClientDuplexStreamImpl.<anonymous> (/Users/repo/pe-consumer/node_modules/salesforce-pubsub-api-client/dist/client.cjs:558:23)
     at ClientDuplexStreamImpl.emit (node:events:513:28)
     at addChunk (node:internal/streams/readable:324:12)
     at readableAddChunk (node:internal/streams/readable:297:9)
     at Readable.push (node:internal/streams/readable:234:10)
     at Object.onReceiveMessage (/Users/repo/pe-consumer/node_modules/@grpc/grpc-js/build/src/client.js:409:24)
[2023-09-20T11:15:11.425Z] Waiting for the debugger to disconnect...
[2023-09-20T11:15:11.419Z] Worker 518d2ced-8ba2-42a2-ba77-2e98fb4c903d exited with code 1
[2023-09-20T11:15:11.554Z] Language Worker Process exited. Pid=8374.
[2023-09-20T11:15:11.554Z] node exited with code 1 (0x1). LanguageWorkerConsoleLog[error] Worker 518d2ced-8ba2-42a2-ba77-2e98fb4c903d uncaught exception (learn more: https://go.microsoft.com/fwlink/?linkid=2097909 ): Error: trailing data
     at Type.fromBuffer (/Users/repo/pe-consumer/node_modules/avro-js/lib/schemas.js:252:11)
     at parseEvent (/Users/repo/pe-consumer/node_modules/salesforce-pubsub-api-client/dist/client.cjs:142:31)
     at /Users/repo/pe-consumer/node_modules/salesforce-pubsub-api-client/dist/client.cjs:559:33
     at Array.forEach (<anonymous>)
     at ClientDuplexStreamImpl.<anonymous> (/Users/repo/pe-consumer/node_modules/salesforce-pubsub-api-client/dist/client.cjs:558:23)
     at ClientDuplexStreamImpl.emit (node:events:513:28)
     at addChunk (node:internal/streams/readable:324:12)
     at readableAddChunk (node:internal/streams/readable:297:9)
     at Readable.push (node:internal/streams/readable:234:10)
     at Object.onReceiveMessage (/Users/repo/pe-consumer/node_modules/@grpc/grpc-js/build/src/client.js:409:24).
[2023-09-20T11:15:11.626Z] Debugger listening on ws://127.0.0.1:9092/7cc2b5e1-a90f-42a6-8258-66386c4c4635
[2023-09-20T11:15:11.626Z] For help, see: https://nodejs.org/en/docs/inspector
[2023-09-20T11:15:11.665Z] Worker f9878be3-cda7-49d3-b1e0-bec28ea548a1 connecting on 127.0.0.1:50463
[2023-09-20T11:15:11.695Z] Worker f9878be3-cda7-49d3-b1e0-bec28ea548a1 received WorkerInitRequest
[2023-09-20T11:15:11.970Z] Loading entry point file "src/functions/subscribe.js"
[2023-09-20T11:15:11.970Z] Setting Node.js programming model to "@azure/functions" version "4.0.0-alpha.12.20230912.4"
[2023-09-20T11:15:11.970Z] Loaded entry point file "src/functions/subscribe.js"
[2023-09-20T11:15:11.970Z] Loading entry point file "src/functions/transformSFevent.js"
[2023-09-20T11:15:11.970Z] Loaded entry point file "src/functions/transformSFevent.js"
[2023-09-20T11:15:11.972Z] Worker process started and initialized.
[2023-09-20T11:15:11.974Z] Worker f9878be3-cda7-49d3-b1e0-bec28ea548a1 received FunctionLoadRequest
[2023-09-20T11:15:11.974Z] Worker f9878be3-cda7-49d3-b1e0-bec28ea548a1 received FunctionLoadRequest
```

**Comment by pozil** [comment]
Hi @bbuyukormeci I'm not sure to understand the error scenario.
Can you describe the steps to reproduce the issue?

**Comment by pozil** [comment]
Hi, I know that this is a bit late but I was able to reproduce the issue and I just released a new version of the client (v3.3.0) that dynamically reloads the event schema if it changes. This means that it will no longer crash if you modify your event in Salesforce.
